### PR TITLE
[11.0][IMP]ao_sale backorder report

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
Solves 21017

Not sure if putting the code in separate module.

### CODE REVIEW

File: ao_sale/migrations/11.0.2.1.0/pre-migration.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    )
+    cr.execute(
+        """
+        UPDATE sale_order_line sol
+        SET qty_to_deliver = 0
+        WHERE sol.product_id in (
+        SELECT pp.id from product_product pp
+        INNER JOIN product_template pt ON pp.product_tmpl_id = pt.id
+        WHERE NOT (pt.invoice_policy = 'delivery'
+        AND pt.service_type = 'manual')
```
Comment (HviorForgeFlow) [inline ao_sale/migrations/11.0.2.1.0/pre-migration.py]: @aheficent Remove last )

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]

**Comment by HviorForgeFlow** [comment]
Solving 21017 but also 20840 ???

File: ao_sale/migrations/11.0.2.1.0/pre-migration.py (hunk 1 above)
Comment (HviorForgeFlow) [reply ao_sale/migrations/11.0.2.1.0/pre-migration.py]: @aheficent not that sorry, last in this line :smile:

**Comment by AaronHForgeFlow** [comment]
> 20840

No. We have to set them manually as delivered. But I think this is a configuration problem in the product. If they don't want to deliver, why don't set the product as "invoice in ordered quantities"?

**Review by HviorForgeFlow — APPROVED** [review]
LGTM :+1: Squash commits needed

**Pull request merged**